quarkus/CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+
+```
+cp -r server/target/lib ${KC_HOME_DIR}
+```
+
+In the example above, the `${KC_HOME_DIR}` variable points to the root directory of the distribution.
+
+You should also be able to update a server dependency directly. For that, copy the jar to the following location:
+
+```
+cp services/target/keycloak-services-${KC_VERSION}.jar ${KC_HOME_DIR}/lib/lib/main/org.keycloak.keycloak-services-${KC_VERSION}.jar
+```
+
+## Running tests
+
+The distribution has its own distribution and the main tests are within the [tests/integration](tests/integration) module.
+
+The test suite has two main types of tests:
+
+* `jvm`
+* `distribution`
+
+The `jvm` tests execute both the test class and server within the same JVM. While the `distribution` tests execute the server
+by running the distribution in a separte JVM.
+
+The `distribution` tests are marked as such using the `DistributionTest` annotation. If not marked with this annotation, the test is a `JVM` test.
+
+To run the tests, execute the following command within the `quarkus` module:
+
+```
+mvn clean install
+```
+
+By default, the tests will run using a raw distribution. If you want to run tests from the [tests/integration](tests/integration) module directly,
+please make sure the distribution was built with the latest changes you did to it and its dependencies.
+
+### Running tests using a container
+
+You can also run tests using a container instead of the raw distribution by setting the `kc.quarkus.tests.dist` property as follows:
+
+```
+mvn clean install -Dkc.quarkus.tests.dist=docker
+```
+
+When setting the `kc.quarkus.tests.dist` to `docker` tests will run using a container instead of the raw distribution.
+
+### Running storage tests
+
+The storage tests are disabled by default but you should be able to run them by activating the `test-database` profile:
+
+```
+mvn clean install -Ptest-database -Dtest=PostgreSQLDistTest
+```
+
+### Running tests from the IDE
+
+You should also be able to run tests from your IDE. For that, choose a test from the [tests/integration](tests/integration) module
+and execute it accordingly.
+
+### Running tests from the `base` testsuite
+
+Sometimes you might want to run the tests from the `base` test suite using the distribution. For that, make sure you have built the `quarkus` module
+and then execute the following command from the project root directory:
+
+```
+mvn -f testsuite/integration-arquillian/pom.xml clean install -Pauth-server-quarkus -Dtest=OIDCProtocolMappersTest
+```
+
+## Documentation
+
+The documentation is a set of guides available from the [docs](../docs/guides/src/main/server) module. Please,
+look at the this [guide](../docs/building.md) about how to update and build the distribution guides.
+
+## Before contributing the changes
+
+Before contributing changes, make to read the main [Keycloak Contributing Guide](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md).
+
+Please, make sure:
+
```

Comment:
Can we also mention the general contributing.md guidelines, especially the issue linking part? -> https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#commit-messages-and-issue-linking

Reply:
There is a link already to the main contributing guide.